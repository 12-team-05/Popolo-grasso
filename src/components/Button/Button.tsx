import React from 'react';
import cn from 'classnames';
import s from './Button.module.scss';

export enum EButtonView {
  default,
  transparent,
}

interface IButtonTProps extends React.HTMLProps<HTMLButtonElement> {
  view?: EButtonView;
  type?: React.ButtonHTMLAttributes<HTMLButtonElement>['type'];
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export default function Button({
  type = 'button', children, className, view = EButtonView.default, ...props
}: IButtonTProps) {
  return (
    <button
      className={cn(s.button, className, {[s.buttonTransparent]: view === EButtonView.transparent})}
      // eslint-disable-next-line react/button-has-type
      type={type}
      {...props}
    >
      {children}
    </button>
  );
}
