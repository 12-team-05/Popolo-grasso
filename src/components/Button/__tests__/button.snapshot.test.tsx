import React from 'react';
import renderer from 'react-test-renderer';
import Button, {EButtonView} from '../Button';

describe('Button snapshots', () => {
  it('renders default button', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders submit button', () => {
    const tree = renderer.create(<Button type="submit" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders submit button', () => {
    const tree = renderer.create(<Button type="reset" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders submit button', () => {
    const tree = renderer.create(<Button view={EButtonView.transparent} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
