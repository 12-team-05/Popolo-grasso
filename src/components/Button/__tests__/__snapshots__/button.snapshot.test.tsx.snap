// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots renders default button 1`] = `
<button
  className="button"
  type="button"
/>
`;

exports[`Button snapshots renders submit button 1`] = `
<button
  className="button"
  type="submit"
/>
`;

exports[`Button snapshots renders submit button 2`] = `
<button
  className="button"
  type="reset"
/>
`;

exports[`Button snapshots renders submit button 3`] = `
<button
  className="button buttonTransparent"
  type="button"
/>
`;
